### ISSUE TITLE
ENT-041: CODEX integration - LoRA fine-tuning for code models

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#10]
## Summary
Support CODEX pipeline output for LoRA fine-tuning of code translation models.

## Tasks
- [ ] Parquet data loader for CODEX output format
- [ ] Code translation loss function (beyond cross-entropy)
- [ ] Multi-task training with rich labels (correctness + similarity + error category)
- [ ] Integration with verificar VER-054 output

## Data Format (from verificar CODEX)
```
source: str          # Python code
target: str          # Rust code  
correct: bool        # Binary correctness
output_similarity: f32   # Soft label [0, 1]
runtime_ratio: f32   # Performance metric
error_category: str  # Multi-class label
```

## Acceptance Criteria
- Load CODEX Parquet directly
- Support soft label distillation
- Multi-task loss combining all label types

## Dependencies
- verificar VER-054 (CODEX pipeline output)
- ENT-037 (KD loss with temperature)

## Ref
verificar: docs/specifications/codex-multi-tech-python-to-rust-spec.md
